### ISSUE TITLE
RFC: Random input generator valid for a grammar defined in .pest file

### ISSUE THREAD

**Issue by laysakura** [pest-parser/pest#503]
### TL;DR

Domain-aware input generator helps fuzzing tests for programs using uses `.pest` files. Same idea with [Testing Random, Valid SQL in CockroachDB](https://www.cockroachlabs.com/blog/testing-random-valid-sql-in-cockroachdb/)

### Motivation

I'm developing a RDBMS whose query language is similar to a subset of SQL. The grammar of the query language is defined in `.pest` file.

My RDBMS has fuzzing tests to check crash-safety with random text inputs, although most inputs are syntactically invalid.
In other words, the test cases can rarely passes lexer, so parsers and backend components are not fully tested.

A fuzzer generating syntactically (and hopefully semantically) valid SQLs would be highly useful to find edge cases human being may not come up with.

Theoretically, such fuzzer (random grammatically-valid text generator) could be implemented when `.pest` file is given. My implementation idea is almost the same as this article: [Testing Random, Valid SQL in CockroachDB](https://www.cockroachlabs.com/blog/testing-random-valid-sql-in-cockroachdb/).

### RFC (Request For Comments)

- Agree with my motivation?
- If such random input generator would be useful, what crate should have the feature?
  - New crate inside this repository?
  - New crate outside of this repository (depending on `pest` crate)?

Thanks,

**Comment by siberianbluerobin** [comment]
I'm not from the pest team, but I did a related project to generate "examples" from pest grammar [Bulk Examples Generator](https://github.com/siberianbluerobin/bulk-examples-generator), not all syntax is supported, but you can try

**Comment by tomtau** [comment]
I think this is a cool idea.

The issue looks to be a duplicate of: https://github.com/pest-parser/pest/issues/209
